samcli/lib/providers/cfn_api_provider.py

```diff
@@ -66,6 +83,9 @@ def extract_resources(self, stacks: List[Stack], collector: ApiCollector, cwd: O
                 if resource_type == AWS_APIGATEWAY_METHOD:
                     self._extract_cloud_formation_method(stack.stack_path, resources, logical_id, resource, collector)
 
+                if resource_type == AWS_APIGATEWAY_AUTHORIZER:
```

Comment:
Nothing specific to this PR, but I would want to us to move to a dispatch pattern whenever there are multiple if branches.

Something like:
```python
for resource_extractor in resource_extractors:
     resource_extract.extract(*args, **kwargs) # pass in the args deemed necessary.
```

might be easier to read and the behavior is apparent.